phosphor-power-supply/power_supply.cpp

```diff
@@ -499,6 +499,25 @@ void PowerSupply::analyzeMFRFault()
     {
         if (mfrFault < DEGLITCH_LIMIT)
         {
+            if (statusMFR & 0x80)
+            {
+                // Report current share warning once per boot
+                if (!currentShareWarningReported)
+                {
+                    log<level::INFO>(
+                        fmt::format("{} 12V current share warning: "
+                                    "STATUS_WORD = {:#06x} "
+                                    "STATUS_MFR_SPECIFIC = {:#04x}",
+                                    shortName, statusWord, statusMFR)
+                            .c_str());
+                    currentShareWarningReported = true;
+                }
+                statusMFR = statusMFR & 0x7F;
+            }
+            if (!statusMFR)
+            {
+                return;
+            }
             if (statusWord != statusWordOld)
```

Comment:
I see two potential problems with these changes, but neither are important enough to not approve the changes.  Just documenting here in case we ever hit them in the field.

1. If a psu first reports a current share fault, then later (10+ sec later) starts reporting both the cs fault and another non-ignored fault (like 12Vcontrolsupply fault), we wouldn't go down this path to trace the statusMFR showing the 12Vcs fault in the journal.  We would still create the serviceable PEL, calling for the psu to be replaced, so I think good enough.
2. If a psu first reported a non-ignored fault (like 12Vcs) we'd correctly create the PEL, but then later if it also started reporting a current share fault, we wouldn't add the journal entry showing the cs fault status.  However, this is no worse than previous code; and our general design is to not have to handle additional faults after already calling out a FRU, so again good enough.

Reply:
I don't understand problem 1 above.  If this bit is set and no other bits in the mfr register, then it exits immediately (not incrementing mfrFault).  If other bits are set, it will continue doing the rest of this method.

However, I might be missing something here, or Faisal might have changed something in his second commit with the change Matt requested.